### PR TITLE
sql: add doc of hex and binary expression support for `LOAD DATA`

### PR DESCRIPTION
**Pull request by imtbkcat**
### What is changed, added or deleted? 
`LOAD DATA` has supported using hex or binary literal to represent ascii character for `ENCLOSED BY` and `TERMINATED BY` in 4.0

This pr add example for `LOAD DATA` doc

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [x] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [ ] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/sql/statements/load-data.md [hunk 1]
```diff
@@ -48,6 +48,14 @@ Query OK, 815264 rows affected (39.63 sec)
 Records: 815264  Deleted: 0  Skipped: 0  Warnings: 0
 ```
 
+`LOAD` `DATA` 也支持使用十六进制 ASCII 字符表达式或二进制 ASCII 字符表达式作为 `FIELDS ENCLOSED BY` 和 `FIELDS TERMINATED BY` 的参数
```
Comment (TomShawn) [inline dev/reference/sql/statements/load-data.md]:
```suggestion
`LOAD DATA` 也支持使用十六进制 ASCII 字符表达式或二进制 ASCII 字符表达式作为 `FIELDS ENCLOSED BY` 和 `FIELDS TERMINATED BY` 的参数。示例如下：
```

File: dev/reference/sql/statements/load-data.md [hunk 2]
```diff
@@ -48,6 +48,14 @@ Query OK, 815264 rows affected (39.63 sec)
...
+
```
Comment (TomShawn) [inline dev/reference/sql/statements/load-data.md]:
```suggestion

{{< copyable "sql" >}}

```

File: dev/reference/sql/statements/load-data.md [hunk 3]
```diff
@@ -48,6 +48,14 @@ Query OK, 815264 rows affected (39.63 sec)
...
+```sql
+LOAD DATA LOCAL INFILE '/mnt/evo970/data-sets/bikeshare-data/2017Q4-capitalbikeshare-tripdata.csv' INTO TABLE trips FIELDS TERMINATED BY x'2c' ENCLOSED BY b'100010' LINES TERMINATED BY '\r\n' IGNORE 1 LINES (duration, start_date, end_date, start_station_number, start_station, end_station_number, end_station, bike_number, member_type);
+```
+
+其中 `x'2c'` 是字符 `,` 的十六进制表示，`b'100010'` 是字符 `"` 的二进制表示
```
Comment (TomShawn) [inline dev/reference/sql/statements/load-data.md]:
```suggestion
以上示例中 `x'2c'` 是字符 `,` 的十六进制表示，`b'100010'` 是字符 `"` 的二进制表示。
```

**Comment by imtbkcat** [comment]
PTAL @TomShawn

**Pull request merged**